File: challenge1.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

class Main {
    public static Scanner potato = new Scanner(System.in);

    static String input(String text) {
        if (text != null) {System.out.println(text);}
        return potato.nextLine();
    }
    public static void main(String[] args) {
        int maxInputs = Integer.parseInt(input(null));
        int radius = maxInputs / 2;
        List<String> array = new ArrayList<>();
        for (int i = 0; i < maxInputs; i++) {
            //System.out.println(i);
            array.add(input(null));
        }
        //String[] halfOfList = array.subList(0, radius).toArray(new String[0]);
        int results = 0;
        for (int i = 0; i < radius; i++) {
            if (array.get(i).equals(array.get(i+radius))) {
                results += 2;
            }
        }
        System.out.println(results);
    }
}

File: challenge_1/challenge1.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

string input() {
    string line;
    getline(cin, line);
    return line;
}

int main() {
    int maxInputs = stoi(input());
    int radius = maxInputs / 2;
    vector<string> array;

    for (int i = 0; i < maxInputs; i++) {
        array.push_back(input());
    }

    int results = 0;
    for (int i = 0; i < radius; i++) {
        if (array[i] == array[i+radius]) {
            results += 2;
        }
    }
    
    cout << results << endl;
    return 0;
}